### ISSUE TITLE
OutOfMemoryException when writing large Spreadsheet

### ISSUE THREAD

**Issue by cmerat** [OfficeDev/Open-XML-SDK#110]
In version 2.6, the Packaging engine was changed. Unfortunately, this new implementation can cause OutOfMemoryExceptions when generating HUGE files. The reason for this is that the use of ZipArchiveEntry (used by ZipArchive that is used by ZipPackage/OpenXMLPackage) when used in Update mode will hold onto all of the data written to it in the UncompressedData stream property. The data is only flushed to disk once the ZipArchiveEntry is closed.

This has had a negative impact for our product with customers experiencing crashes in production and we've had to revert to version 2.5 until this issue is resolved.

Admittedly, generation of extremely large workbooks is not a typical use case but this is the exact reason why we chose the OpenXML over other existing 3rd party Excel libraries.

**Comment by tarunchopra** [comment]
@cmerat : Can you please share a sample code to repro. the issue?

**Comment by tarunchopra** [comment]
@cmerat: Any update please?

**Comment by cmerat** [comment]
@tarunchopra The repro case was in a production scenario with a customer along with our application code so obviously I can't just send this directly over. I'll try to get a sample that reproduces the case to you asap but it might take a few days.

**Comment by tarunchopra** [comment]
@cmerat : Sounds good; thx

**Comment by cmerat** [comment]
You'll find attached the repro case. The NuGet package is already set to cause the problem, running the executable you'll quickly see rising memory usage. You can manage the NuGet package to downgrade to 2.5.1 and the problem is gone, memory usage remains stable during execution.

Note that this will create a huge (1GB+) xlsx file at c:\temp\Output.xlsx. Make sure you have a temp folder and enough space available.

[OpenXmlMemoryIssue.zip](https://github.com/OfficeDev/Open-XML-SDK/files/570092/OpenXmlMemoryIssue.zip)

**Comment by tarunchopra** [comment]
Thx, I will have a look.

**Comment by tarunchopra** [comment]
@cmerat: Thanks for the help and repro code. I tested your code with new binary set ( #112 which removed our custom System.IO.Packaging.) and didn't find a memory leak. Would request to please retest your code with latest binary set.

**Comment by cmerat** [comment]
@tarunchopra forgot to get back to you but it indeed appears to be fixed in the new binary set. Thanks!

**Comment by cmerat** [comment]
@tarunchopra this issue is back when using DotNet Core with the System.IO.Packaging 4.4.0. There is an issue open about it here: https://github.com/dotnet/corefx/issues/24457

Currently, this is preventing us from creating very large spreadsheets in DotNet Core.

**Comment by tarunchopra** [comment]
@cmerat : Unfortunately; we can't do much until the issue is fixed by corxfx team.

**Comment by cmerat** [comment]
Apparently, one of the reasons this hasn't been addressed yet is that it is considered a low priority improvement. At the very least, informing the corefx team of the impact of this on the OpenXml package may put some additional weight to the request.

**Comment by tarunchopra** [comment]
@twsouthwick : Do you have any input on @cmerat comment?

**Comment by twsouthwick** [comment]
The issue is currently blocked by .NET Core not having a support for streaming zip files. The more the community lets the team know, the more likely it is to be addressed. The issue you referenced above is a good way to get more eyes on it :)
